python/ray/serve/deployment_state.py

```diff
@@ -1093,13 +1093,13 @@ def deploy(self, deployment_info: DeploymentInfo) -> bool:
         # NOTE(edoakes): we must write a checkpoint before starting new
         # or pushing the updated config to avoid inconsistent state if we
         # crash while making the change.
-        self._save_checkpoint_func()
+        await self._save_checkpoint_func()
 
         return True
 
-    def delete(self) -> None:
+    async def delete(self) -> None:
```

Comment:
Could we add the thread-safety comment to this function? The user may have to retry their delete requests if the controller crashes before checkpointing.